tsl/src/bgw_policy/compression_api.c

```diff
@@ -347,6 +374,19 @@ policy_compression_add_internal(Oid user_rel_oid, Datum compress_after_datum,
 Datum
 policy_compression_add(PG_FUNCTION_ARGS)
 {
+	Oid fn_oid = FC_FN_OID(fcinfo);
+	bool is_columnstore_policy = false;
+
+	/*
+	 * `policy_compression_add` is used by both `add_compression_policy()` and
+	 * `add_columnstore_policy()`; but thier arguments names differ. We need to
+	 * respect that in the error messages.
+	 */
+	if (OidIsValid(fn_oid) && strcmp(get_func_name(fn_oid), "add_columnstore_policy") == 0)
+	{
+		is_columnstore_policy = true;
+	}
```

Comment:
I could have added two different C functions for `policy_compression_add` and `add_compression_policy` to differentiate between them but I thought it would be excessive. Instead I check which SQL function invoked the call.